dspace-api/src/main/java/org/dspace/identifier/HandleIdentifierProvider.java

```diff
@@ -50,21 +51,30 @@ public boolean supports(Class<? extends Identifier> identifier) {
     @Override
     public boolean supports(String identifier)
     {
-        String prefix = handleService.getPrefix();
-        String handleResolver = ConfigurationManager.getProperty("handle.canonical.prefix");
+    	String prefix = handleService.getPrefix();
+        String canonicalPrefix = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("handle.canonical.prefix");
         if (identifier == null)
         {
             return false;
         }
+        // return true if handle has valid starting pattern
         if (identifier.startsWith(prefix)
-                || identifier.startsWith(handleResolver)
-                || identifier.startsWith("http://hdl.handle.net/")
+                || identifier.startsWith(canonicalPrefix)
                 || identifier.startsWith("hdl:")
-                || identifier.startsWith("info:hdl"))
+                || identifier.startsWith("info:hdl")
+                || identifier.matches("^https?://hdl\\.handle\\.net/.*")
+                || identifier.matches("^https?://.+/handle/.*"))
         {
             return true;
         }
-        
+        // return true if base prefix matches in case of multi-instance deployment with derived prefixes demarcated by a dot "." 
+        if(prefix.contains(".")) {
+            String[] splitPrefix = prefix.split("\\.");
+            if(splitPrefix.length > 1 && identifier.startsWith(splitPrefix[0])) {
+                return true;
+            }
+        }
```

Comment:
I just realized this logic may be slightly flawed for brand new Handle prefixes.  For those who haven't heard, when you buy a new prefix from the Handle.net Registry, they are now of the format `20.500.xxxxx` (i.e ALL handle prefixes now begin with "20.500.*").  This means that newer "derived" prefixes will end up being of the format `20.500.xxxx.yyy`, see this FAQ question on their site for an example: https://www.handle.net/faq.html#4

Unfortunately, the logic above may not work right for new prefixes, as it seems to assume there will only be *one* period within the prefix.  To be *more correct* this probably should be matching against the *last period* in the prefix.. something like

```
       if(prefix.contains(".")) {
            String prefixBeforeLastPeriod = prefix.substring(0, prefix.lastIndexOf(".")+1);
            if(prefixBeforeLastPeriod.length > 1 && identifier.startsWith(prefixBeforeLastPeriod)) {
                return true;
            }
        }
```
This still isn't "perfect" , as a configuring a prefix of "20.500.1000" will also cause a prefix of "20.500.88" to return "true".  But, I'm not sure there's any way to make this 100% perfect.